lib/rialto/etl/configs/stanford_people_to_sparql_statements.rb

```diff
@@ -41,70 +50,123 @@
-  # hardcoding to US (http://sws.geonames.org/6252001/)
-  to_field VCARD['country-name'].to_s, literal('United States'), single: true
-  to_field DCTERMS['spatial'].to_s, literal(RDF::URI.new('http://sws.geonames.org/6252001/')), single: true
-end
+to_field "!#{VCARD['hasAddress']}", literal(true), single: true
+to_field VCARD['hasAddress'].to_s, lambda { |json, accum|
+  address_json = JsonPath.on(json, '$.contacts[?(@["type"] == "academic")]').first
+  if address_json
+    accum << {
+      '@id' => RIALTO_CONTEXT_ADDRESSES[json['profileId']],
+      '@type' => VCARD['Address'],
+      "!#{VCARD['street-address']}" => true,
+      VCARD['street-address'].to_s => address_json['address'],
+      "!#{VCARD['locality']}" => true,
+      VCARD['locality'].to_s => address_json['city'],
+      "!#{VCARD['region']}" => true,
+      VCARD['region'].to_s => address_json['state'],
+      "!#{VCARD['postal-code']}" => true,
+      VCARD['postal-code'].to_s => address_json['zip'],
+      # Punting on looking up country based on postal code (http://www.geonames.org/export/web-services.html) and
+      # hardcoding to US (http://sws.geonames.org/6252001/)
+      "!#{VCARD['country-name']}" => true,
+      VCARD['country-name'].to_s => 'United States',
+      "!#{DCTERMS['spatial']}" => true,
+      DCTERMS['spatial'].to_s => RDF::URI.new('http://sws.geonames.org/6252001/')
+    }
+  end
+}, single: true
 
-# Advisees
-# Note that advisees are not deleted.
-to_field '@advisees', lambda { |json, accum|
-  advisees_json = JsonPath.on(json, '$.advisees')
-  unless advisees_json.empty?
-    advisees_json[0].each do |advisee_json|
-      advisee_hash = {}
-      advisee_hash['@id'] = JsonPath.on(advisee_json, '$.advisee.profileId').first
-      advisee_hash['@id_ns'] = RIALTO_PEOPLE.to_s
-      advisee_hash['@type'] = [FOAF['Agent'], FOAF['Person']]
-      advisee_hash['@label'] = [JsonPath.on(advisee_json, '$.advisee.firstName').first,
-                                JsonPath.on(advisee_json, '$.advisee.lastName').first].join(' ')
-      advisee_hash['!type'] = true
-      name_vcard_hash = {}
-      name_vcard_hash[VCARD['given-name'].to_s] = JsonPath.on(advisee_json, '$.advisee.firstName').first
-      name_vcard_hash[VCARD['family-name'].to_s] = JsonPath.on(advisee_json, '$.advisee.lastName').first
-      advisee_hash['@person_name'] = [name_vcard_hash]
-      advisee_hash['!person_name'] = true
-      accum << advisee_hash
-    end
+# Note: There is also relatedBy for positions.
+to_field VIVO['relatedBy'].to_s, lambda { |json, accum|
+  advisees_json = JsonPath.on(json, '$.advisees[*].advisee')
+  advisees = []
+  advisees_json.each do |advisee_json|
+    advisees << {
+      '@id' => RIALTO_CONTEXT_RELATIONSHIPS["#{advisee_json['profileId']}_#{json['profileId']}"],
+      '@type' => VIVO['AdvisingRelationship'],
+      DCTERMS['valid'].to_s => Time.now.to_date
+    }
   end
+  accum.concat(advisees)
```

Comment:
Did you choose `#concat` over `<<` here intentionally? If we can use `<<` here, I have a slight preference for it to increase internal consistency.

Reply:
Intentionally. Since VIVO.relatedBy is added to by advisees and positions, you get `[[advisee1, advisee2], [position1, position2]]` with `<<`, when what you want is `[advisee1, advisee2, position1, position2]` which you get from `concat`.

This is another instance where Traject just gets in the way.